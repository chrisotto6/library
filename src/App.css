.App {
  font-family: 'Trebuchet MS', Helvetica, sans-serif;
  background-color: white;
  height: 100vh;
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

.App-header {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  padding: 50px 0 30px 0;
  color: #282c34;
}
