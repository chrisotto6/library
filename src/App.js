import React from 'react'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { Footer, Navigation, RecentBooks, ToReadBooks, HighlyRated, StatsPage } from './components'
import './App.css'

function App() {
  return (
    <div className="App">
      <Router>
        <Navigation />
        <Switch>
          <Route path="/" exact component={() => <RecentBooks />} />
          <Route path="/to-read" exact component={() => <ToReadBooks />} />
          <Route path="/highly-rated" exact component={() => <HighlyRated />} />
          <Route path="/stats" exact component={() => <StatsPage />} />
        </Switch>
        <Footer />
      </Router>
    </div>
  )
}

export default App
