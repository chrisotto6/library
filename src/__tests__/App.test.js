import React from 'react'
import renderer from 'react-test-renderer'
import App from '../App'

test('App Snapshot Matches', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
