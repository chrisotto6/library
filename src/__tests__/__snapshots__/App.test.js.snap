// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Matches 1`] = `
<div
  className="App"
>
  <div
    className="navigation"
  >
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-secondary justify-content-between"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Library
        </a>
        <button
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarResponsive"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarResponsive"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item  active"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Recently Read
                <span
                  className="sr-only"
                >
                  (current)
                </span>
              </a>
            </li>
            <li
              className="nav-item  "
            >
              <a
                className="nav-link"
                href="/to-read"
                onClick={[Function]}
              >
                To Read
              </a>
            </li>
            <li
              className="nav-item  "
            >
              <a
                className="nav-link"
                href="/highly-rated"
                onClick={[Function]}
              >
                Highly Rated
              </a>
            </li>
            <li
              className="nav-item  "
            >
              <a
                className="nav-link"
                href="/stats"
                onClick={[Function]}
              >
                Stats
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="booksContainer"
  >
    <p
      className="loading"
    >
      Loading books from Goodreads...
    </p>
    <ul />
  </div>
  <div
    className="footer"
  >
    <footer
      className="py-4 navbar-dark bg-secondary"
    >
      <div
        className="container"
      >
        <p
          className="m-0 text-center text-white"
        >
          Copyright Â© Chris Otto
          2021
        </p>
      </div>
    </footer>
  </div>
</div>
`;
